service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java

```diff
@@ -0,0 +1,82 @@
+package io.smallrye.stork.servicediscovery.kubernetes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.stork.ServiceDiscovery;
+import io.smallrye.stork.ServiceInstance;
+import io.smallrye.stork.config.ServiceDiscoveryConfig;
+import io.smallrye.stork.spi.ServiceInstanceIds;
+
+public class KubernetesServiceDiscovery implements ServiceDiscovery {
+
+    private final KubernetesClient client;
+    private final String serviceName;
+    private boolean allNamespaces = false;
+    private String namespace;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KubernetesServiceDiscovery.class);
+
+    public KubernetesServiceDiscovery(String serviceName, ServiceDiscoveryConfig config) {
+        Config base = Config.autoConfigure(null);
+        this.serviceName = serviceName;
+        Map<String, String> parameters = config.parameters();
+        String masterUrl = parameters.get("k8s-host");
+        if (masterUrl == null) {
+            masterUrl = base.getMasterUrl();
+        }
+        namespace = parameters.get("k8s-namespace");
+        if (namespace == null) {
+            namespace = base.getNamespace();
+        } else if (namespace.equalsIgnoreCase("all")) {
+            allNamespaces = true;
+        }
+        Config properties = new ConfigBuilder(base)
+                .withMasterUrl(masterUrl)
+                .withNamespace(namespace).build();
+        client = new DefaultKubernetesClient(properties);
+    }
+
+    public Uni<List<ServiceInstance>> getServiceInstances() {
+        List<Endpoints> endpoints = allNamespaces
```

Comment:
`client.endpoints...` is a blocking call...
We should either find a way to make it non-blocking (if the client supports it) or wrap it in `vertx.executeBlocking`. The latter would require adding vert.x as a dependency but I think it's okay.

Reply:
Ok, I will take a look